src/ranged.cpp

```diff
@@ -247,7 +247,6 @@ dealt_projectile_attack Creature::projectile_attack( const projectile &proj_arg,
             // Don't do anything, especially don't call map::shoot as this would damage the vehicle
         } else {
             g->m.shoot( tp, proj, !no_item_damage && tp == target_arg );
-            has_momentum = proj.impact.total_damage() > 0;
```

Comment:
Nevermind, I see why it's related, but this is still wrong, this code allows the projectile to spend some of its momentum breaking through relatively fragile map features like thin doors and windows and continuing to hit something else, this change means projectiles will keep going through any map features such as walls, even if it is unable to destroy them.
Concrete example, you can throw a grenade through a wall with this line removed.